compiler/rustc_resolve/src/lib.rs

```diff
@@ -3392,6 +3403,38 @@ impl<'a> Resolver<'a> {
         }
         None
     }
+
+    fn resolve_main_by_path(&mut self, krate: &ast::Crate) {
+        let any_exe = self.session.crate_types().iter().any(|ty| *ty == CrateType::Executable);
+        if !any_exe {
+            // No need to find a main function.
+            return;
+        }
+
+        // If the user wants no main function at all, then stop here.
+        if self.session.contains_name(&krate.attrs, sym::no_main) {
+            return;
+        }
```

Comment:
This condition is irrelevant here, already checked in `fn entry_fn` and can be removed.

Reply:
Done.